test/versioned/when/segments.test.js

```diff
@@ -0,0 +1,448 @@
+      assertSegments(
+        tx.trace.root,
+        [
+          'doWork1',
+          [
+            'Promise startSomeWork',
+            [
+              'Promise#then firstThen',
+              [
+                'doWork2',
+                ['Promise startSomeWork', ['Promise#then <anonymous>', 'Promise#then secondThen']]
+              ]
+            ]
+          ]
+        ],
+        { exact: false },
+        { assert: plan }
+      )
+    })
+
+    helper.runInTransaction(agent, function transactionWrapper(transaction) {
+      doSomeWork({ tracer, segmentName: 'doWork1', Promise: when.Promise })
+        .then(function firstThen() {
+          return doSomeWork({ tracer, segmentName: 'doWork2', Promise: when.Promise })
+        })
+        .then(function secondThen() {
+          const s = tracer.createSegment('secondThen')
+          s.start()
+          s.end()
+          process.nextTick(transaction.end.bind(transaction))
+        })
+    })
+
+    await plan.completed
+  })
+
+  await t.test('then handler that returns a value', async (t) => {
+    const plan = tspl(t, { plan: 10 })
+    const { agent, tracer, when } = t.nr
+
+    agent.once('transactionFinished', function (tx) {
+      plan.equal(tx.trace.root.children.length, 1)
+
+      assertSegments(
+        tx.trace.root,
+        [
+          'doWork1',
+          ['Promise startSomeWork', ['Promise#then firstThen', ['Promise#then secondThen']]]
+        ],
+        {},
+        { assert: plan }
+      )
+    })
+
+    helper.runInTransaction(agent, function transactionWrapper(transaction) {
+      doSomeWork({ tracer, segmentName: 'doWork1', Promise: when.Promise })
+        .then(function firstThen() {
+          return 'some value'
+        })
+        .then(function secondThen() {
+          process.nextTick(transaction.end.bind(transaction))
+        })
+    })
+
+    await plan.completed
+  })
+
+  await t.test('catch handler with error from original promise', async (t) => {
+    const plan = tspl(t, { plan: 8 })
+    const { agent, tracer, when } = t.nr
+
+    agent.once('transactionFinished', function (tx) {
+      plan.equal(tx.trace.root.children.length, 1)
+
+      assertSegments(
+        tx.trace.root,
+        ['doWork1', ['Promise startSomeWork', ['Promise#catch catchHandler']]],
+        {},
+        { assert: plan }
+      )
+    })
+
+    helper.runInTransaction(agent, function transactionWrapper(transaction) {
+      doSomeWork({ tracer, segmentName: 'doWork1', shouldReject: true, Promise: when.Promise })
+        .then(function firstThen() {
+          return 'some value'
+        })
+        .catch(function catchHandler() {
+          process.nextTick(transaction.end.bind(transaction))
+        })
+    })
+
+    await plan.completed
+  })
+
+  // Note: this test was originally a `promise_segments: false` test listed
+  // under the segments enabled suite.
+  await t.test('catch handler with error from subsequent promise', async (t) => {
```

Comment:
This one was originally https://github.com/newrelic/node-newrelic/blob/dffdac7f450a23c6e23a173c91bda583d40e573d/test/versioned/when/legacy-promise-segments.js#L144-L156

It also should not have been under `promise_segments: true`.